### PR TITLE
feat(schematics): added ng-add schematics command

### LINKED ISSUE #26: schematics: create ng-add

### PR DESCRIPTION
**Pull request by itayod**
Finally ^_^

added schematics that add the following:
* translation files by user input
* http loader by user input
* add `TranslocoModule` to module declarations
* add config to module providers

## TODO:
[ ] tests.
[ ] use environment in config.

close #26

### CODE REVIEW

File: schematics/src/collection.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "$schema": "../node_modules/@angular-devkit/schematics/collection-schema.json",
+  "schematics": {
+    "ng-add": {
+      "aliases": ["init"],
+      "factory": "./ng-add/",
+      "schema": "./ng-add/schema.json",
+      "description": "Add root side effect class"
```
Comment (NetanelBasal) [inline schematics/src/collection.json]: description is wrong (copy paste 😂)

File: schematics/src/ng-add/index.ts [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+    return applyChanges(host, options.module, [change]);
+  };
+}
+
+function getLoaderTemplates(loader: Loaders): { loaderTemplate: string; loaderProvider: string } {
+  const httpTemplate = `
+export function HttpLoader(http: HttpClient) {
+  return function(lang: string) {
+    return http.get('/assets/i18n/' + lang + '.json');
```
Comment (NetanelBasal) [inline schematics/src/ng-add/index.ts]: use template strings

File: schematics/src/ng-add/index.ts [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+  };
+}`;
+
+  const webpackTemplate = `
+export function WebpackLoader() {
+  return function(lang: string) {
+    return import('../assets/langs/' + lang + '.json');
```
Comment (NetanelBasal) [inline schematics/src/ng-add/index.ts]: use template strings

Comment (NetanelBasal) [reply schematics/src/ng-add/index.ts]: change langs to i18n

Comment (NetanelBasal) [reply schematics/src/ng-add/index.ts]: add `.then(module => module.default);`

File: schematics/src/ng-add/index.ts [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+      return { loaderTemplate: webpackTemplate, loaderProvider: provider('WebpackLoader') };
+  }
+}
+
+export default function(options: SchemaOptions): Rule {
+  return (host: Tree, context: SchematicContext) => {
+    const langs = options.langs.split(',').map(l => l.trim());
+    const project = getProject(host);
+
+    // TODO: try not to taking it as HC.
```
Comment (NetanelBasal) [inline schematics/src/ng-add/index.ts]: what is HC?

File: schematics/src/ng-add/index.ts [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+        defaultLang: 'en'
+      } as TranslocoConfig
+    }`;
+    let { loaderTemplate, loaderProvider } = getLoaderTemplates(options.loader);
+
+    return chain([
+      options.loader === Loaders.Http
+        ? addImportsToModuleFile(options, ['HttpClient'], '@angular/common/http')
+        : noop(),
+      // mergeWith(translateFiles),
```
Comment (NetanelBasal) [inline schematics/src/ng-add/index.ts]: If you don't need it, remove it.

File: schematics/src/ng-add/schema.json [hunk 6]
```diff
@@ -0,0 +1,29 @@
+{
+  "$schema": "http://json-schema.org/schema",
+  "id": "SchematicsTranslocoAdd",
+  "title": "Add Transloco to Your Project",
+  "type": "object",
+  "properties": {
+    "langs": {
+      "description": "The languages of the project.",
+      "x-prompt": "Which languages do you need?",
```
Comment (NetanelBasal) [inline schematics/src/ng-add/schema.json]: emojis!!

File: schematics/src/ng-add/index.ts [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+import { SchemaOptions, Loaders } from './schema';
+
+function createTranslateFiles(langs: string[]): HostTree {
+  const treeSource = new EmptyTree();
+  langs.forEach(lang => {
+    treeSource.create(
+      lang + '.json',
+      `
+      {
+
```
Comment (NetanelBasal) [inline schematics/src/ng-add/index.ts]: Add one example key: `"hello": "transloco currentLanguage"`

File: schematics/src/ng-add/index.ts [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+    // TODO: try not to taking it as HC.
+    const assetsPath = project.root + `src/assets/i18n/`;
+    const translateFiles = apply(source(createTranslateFiles(langs)), [move('/', assetsPath)]);
+
+    options.module = findRootModule(host, options.module, project.sourceRoot) as string;
+    const configProviderTemplate = `{
+      provide: TRANSLOCO_CONFIG,
+      useValue: {
+        runtime: true,
```
Comment (NetanelBasal) [inline schematics/src/ng-add/index.ts]: Add `prodMode`:
```
{
   prodMode: environment.production,
}
```

Comment (NetanelBasal) [inline schematics/src/ng-add/index.ts]: runtime false by default

Comment (shaharkazaz) [reply schematics/src/ng-add/index.ts]: @NetanelBasal in our default config it's true, should we change to false?

Comment (NetanelBasal) [reply schematics/src/ng-add/index.ts]: Yes.

File: schematics/src/collection.json (hunk 1 above)
Comment (itayod) [reply schematics/src/collection.json]: Hey, it's my code so I allow ;)

File: schematics/src/ng-add/index.ts (hunk 4 above)
Comment (itayod) [reply schematics/src/ng-add/index.ts]: hard coded

File: schematics/src/ng-add/index.ts (hunk 8 above)
Comment (shaharkazaz) [reply schematics/src/ng-add/index.ts]: The specs 😭

File: schematics/src/ng-add/index.ts (hunk 5 above)
Comment (itayod) [reply schematics/src/ng-add/index.ts]: oops left it marked out for testing

File: schematics/src/ng-add/index.ts (hunk 4 above)
Comment (NetanelBasal) [reply schematics/src/ng-add/index.ts]: Yes, that's a good point, but we can leave it for later.

File: schematics/src/ng-add/index.ts (hunk 8 above)
Comment (NetanelBasal) [reply schematics/src/ng-add/index.ts]: wallak ata

**Pull request merged**